pkg/kubelet/kubelet.go

```diff
@@ -771,6 +777,43 @@ func (kl *Kubelet) makeEnvironmentVariables(ns string, container *api.Container)
 	return result, nil
 }
 
+// runtimeEnvVarValue determines the value that an env var should take when a container
+// is started.  If the value of the env var is the empty string, the source of the env var
+// is resolved, if one is specified.
+//
+// TODO: preliminary factoring; make better
+func (kl *Kubelet) runtimeEnvVarValue(envVar api.EnvVar, pod *api.Pod) (string, error) {
+	runtimeVal := envVar.Value
+	if runtimeVal == "" {
+		switch {
+		case envVar.ObjectInfo != nil:
+			objInfo := envVar.ObjectInfo
+			internalFieldPath, _, err := api.Scheme.ConvertFieldLabel(objInfo.APIVersion, "Pod", objInfo.FieldPath, "")
```

Comment:
add the `TODO: use Kind from ObjectInfo` so it doesn't confuse people looking at this.